### PR TITLE
Fix tracking replicaSet: ensure desired replicas is > 0

### PR DESCRIPTION
**Pull request by tobiscr**

### CODE REVIEW

**Review by pbochynski — APPROVED** [review]

File: pkg/reconciler/kubernetes/progress/ready.go [hunk 1]
```diff
@@ -120,15 +121,45 @@ func getLatestReplicaSet(ctx context.Context, deployment *appsv1.Deployment, cli
...
+	t1Copy := template1.DeepCopy()
+	t2Copy := template2.DeepCopy()
+	// Remove hash labels from template.Labels before comparing
+	delete(t1Copy.Labels, appsv1.DefaultDeploymentUniqueLabelKey)
+	delete(t2Copy.Labels, appsv1.DefaultDeploymentUniqueLabelKey)
+	return apiequality.Semantic.DeepEqual(t1Copy, t2Copy)
+}
+
+// ReplicaSetsByCreationTimestamp sorts a list of ReplicaSet by creation timestamp, using their names as a tie breaker.
+type ReplicaSetsByCreationTimestamp []*appsv1.ReplicaSet
```
Comment (skhalash) [inline pkg/reconciler/kubernetes/progress/ready.go:158]: Small hint - it does not have to be exported

File: pkg/reconciler/kubernetes/progress/ready.go [hunk 2]
```diff
@@ -120,15 +121,45 @@ func getLatestReplicaSet(ctx context.Context, deployment *appsv1.Deployment, cli
...
 	}
 
-	sort.Sort(replicaSetsByCreationTimestamp(ownedReplicaSets))
-	return ownedReplicaSets[len(ownedReplicaSets)-1], nil
+	return FindNewReplicaSet(deployment, ownedReplicaSets), nil
 }
 
-type replicaSetsByCreationTimestamp []*appsv1.ReplicaSet
+// FindNewReplicaSet returns the new RS this given deployment targets (the one with the same pod template).
+func FindNewReplicaSet(deployment *appsv1.Deployment, rsList []*appsv1.ReplicaSet) *appsv1.ReplicaSet {
```
Comment (skhalash) [inline pkg/reconciler/kubernetes/progress/ready.go:128]: This one should not be exported either :)

Comment (tobiscr) [reply pkg/reconciler/kubernetes/progress/ready.go]: fully agree, fixed

File: pkg/reconciler/kubernetes/progress/ready.go (hunk 1 above)
Comment (tobiscr) [reply pkg/reconciler/kubernetes/progress/ready.go]: fixed

**Review by skhalash — APPROVED** [review]

**Review by clebs — APPROVED** [review]

**Pull request merged**